### PR TITLE
support setting WebSocket Base URL

### PR DESCRIPTION
**Pull request by asamere**
Would be better to create an instance to manage the websocket interactions as is done for the rest client. this is gets it done on my end.

### CODE REVIEW

File: v2/websocket_service.go [hunk 1]
```diff
@@ -9,33 +9,43 @@ import (
...
+func init() {
 	if UseTestnet {
-		return baseWsTestnetURL
+		baseURL = baseTestnetURL
 	}
+	SetWebsocketBaseURL(baseURL)
+}
+
+// failing to set the websocket base url prior to use can lead to
+// interesting scenarios syncing an order bool when the rest client
```
Comment (asamere) [inline v2/websocket_service.go:34]: book

File: v2/websocket_service.go [hunk 2]
```diff
@@ -9,33 +9,43 @@ import (
...
+	baseTestnetURL = "wss://testnet.binance.vision"
+	// derived from baseURL
+	baseWsMainURL       string
+	baseCombinedMainURL string
 )
 
-// getWsEndpoint return the base endpoint of the WS according the UseTestnet flag
-func getWsEndpoint() string {
+func init() {
 	if UseTestnet {
```
Comment (norunners) [inline v2/websocket_service.go:27]: Do not use init with a mutable global. It's likely to run before `UseTestnet` is set.

Comment (adshao) [reply v2/websocket_service.go:27]: hi @asamere , check this comment, I agree with @norunners . `UseTestnet` could be updated after `init`.

**Comment by xyq-c-cpp** [comment]
No response for a long time, close this.

**Pull request closed without merging**